ratelimits/names.go

```diff
@@ -146,7 +147,7 @@ func validateRegId(id string) error {
 // where domain is a domain name.
 func validateDomain(id string) error {
 	err := policy.ValidDomain(id)
-	if err != nil {
+	if err != nil && !errors.Is(err, policy.ErrWildcardNotSupported) {
```

Comment:
This feels a little weird and abstraction-violation-y to me. And I think the root of it is that `policy.ValidDomain` isn't really what we want to be using here. That function is designed for a specific purpose: are we allowed to, _per our regulatory requirements_, issue for this domain? That's pretty different from what I think this function really wants, which is: does this string look like a domain name?

See also https://github.com/letsencrypt/boulder/pull/7212, where the fact that policy.ValidDomain is exported makes me slightly nervous.

Maybe a cleaner check would be using net/url.ParseURL and requiring that everything except for .Host is empty?

Reply:
After looking at this long enough in #7218, I realize now that when a wildcard is present we're skipping a lot of other validation that should probably take place as well. Lets see how #7218 shakes out, if necessary, we can always copy over ValidWildcardDomain() from that PR into this one.

Reply:
It's not clear to me why the ratelimits layer needs to validate the format of domain names at all. Validation has already been done at a different layer. Sometimes belt-and-suspenders checks are good, but as far as I can tell nothing will go wrong if the ratelimits code somehow receives a malformed domain name.

I would recommend just deleting this check if it's causing abstraction violation problems.